pkg/kdump/config-client.es6

```diff
@@ -0,0 +1,165 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from 'cockpit';
+
+/* Parse an ini-style config file
+ * and monitor it for changes
+ */
+export class ConfigFile {
+    constructor(filename, superuser = false) {
+        this.filename = filename;
+        this._rawContent = undefined;
+        this._lines = [ ];
+        this._originalSettings = { };
+        this._dataAvailable = cockpit.defer();
+        this.settings = { };
+
+        cockpit.event_target(this);
+
+        this._fileHandle = cockpit.file(filename, { superuser: superuser } );
+        var self = this;
+        this._fileHandle.watch(function(rawContent) {
+            self._parseText(rawContent);
+        });
+    }
+    close() {
+        if (this._fileHandle) {
+            this._fileHandle.remove();
+            this._fileHandle = undefined;
+        }
+    }
+    // wait for data to have been read at least once
+    wait() {
+        return this._dataAvailable.promise();
+    }
+    /* parse lines of the config file
+     * if a line has a valid config key, use that as key
+     * and also store original line, line index, value and whether the line contains a comment
+     * we care about the comment since we don't want to overwrite a user comment when changing a value
+     * e.g. for line "someKey foo # comment"
+     * outputObject["someKey"] = { index: 0, value: "foo", origLine: "someKey foo # comment", hasComment: true }
+     * skipNotify: Don't notify about changes, e.g.to avoid multiple updates when writing a file
+     */
+    _parseText(rawContent, skipNotify) {
+        if (this._dataAvailable)
+            this._dataAvailable.resolve();
+
+        var self = this;
```

Comment:
Is this really necessary in ES6? Shouldn't ```this``` be more reliable and stably accessible?